packages/kit/types/index.d.ts

```diff
@@ -340,6 +340,16 @@ export interface KitConfig {
 		 */
 		checkOrigin?: boolean;
 	};
+	/**
+	 * Here be dragons. Enable at your peril.
+	 */
+	dangerZone?: {
+		/**
+		 * Automatically add server-side `fetch`ed URLs to the `dependencies` map of `load` functions. This will expose secrets
+		 * to the client if your URL contains them.
+		 */
+		trackServerFetchesPotentiallyExposingSecrets?: boolean;
```

Comment:
I think we can trim the option name down to `trackServerFetches` — we can explain the details in the inline documentation, and the `dangerZone` makes it clear that this is 'know what you're doing' territory

```suggestion
		/**
		 * Mark server `load` functions that call `fetch(url)` as depending on `url`, allowing you to
		 * call `invalidate(url)` from the client. The URL, including e.g. sensitive query parameters,
		 * will be visible to the client. This option will be removed in version 2; use `depends` instead
		 */
		trackServerFetches?: boolean;
```

Reply:
I dunno, I like explicitly requiring users to acknowledge what they're doing -- there's no guarantee they understand why this is dangerous. Happy to make the change, but I personally think the annoyance of typing a gigantic and explicit config value is worth it, especially since it's not going to be around forever.